### PR TITLE
Implement feature to open folder and show its content

### PR DESCRIPTION
**Pull request by BohdanDudkaTD**
This PR adds the possibility to open the folder by click on it. After click `fetch` action is dispatched with the id of the folder whose content should be displayed.

Result of the ESLint execution:
image

### CODE REVIEW

**Comment by BohdanDudkaTD** [comment]
@dkalinin, PTAL

File: app/view/component/file-list-item/index.js [hunk 1]
```diff
@@ -143,6 +143,13 @@ export class FileListItem extends DragAndDropComponent {
       this._listItem.addEventListener('dragenter', (event) => {
         this.onDragEnter(event, this._listItem);
       });
+
+      this._listItem.addEventListener('click', (event) => {
+        event.preventDefault();
+        event.stopPropagation();
+        this.dispatch(actions.FETCH_ITEMS, this._file.id)
+            .catch((error) => alert(error));
```
Comment (dkalinin) [inline app/view/component/file-list-item/index.js]: Please show name of current folder.

**Comment by BohdanDudkaTD** [comment]
@dkalinin, PTAL, fixed issues:
- log id of the folder which was opened;
- store `id of the current folder` in the state - it's done by `SET_CURRENT_FOLDER_ID` mutation, which is called when items were successfully fetched.

File: app/view/component/configurable-component/index.js [hunk 2]
```diff
@@ -20,6 +20,21 @@ export class ConfigurableComponent {
     return this._store.state;
   }
 
+  /**
+   * Gets the id of the currently opened folder. If it's undefined or equals to my-drive - return
+   * value my-drive. Otherwise, return id of the currently opened folder from the state.
+   *
+   * @return {string} id of the currently opened folder.
+   */
+  get currentFolderId() {
```
Comment (dkalinin) [inline app/view/component/configurable-component/index.js]: Please remove from Configurable component.

**Comment by BohdanDudkaTD** [comment]
@dkalinin, PTAL, fixed issues:
- get root folder at first render of the `FileListPage`. `FileList` will be rendered only if files were fetched successfully;
- now the sequence of actions is:
1.  load `root\current` folder;  
2.  set it as a `currentFolder`;  
3. fetch its items.

- remove unnecessary `currentFolderId` method from configurable component;
- in a few methods of the `FileRequestService` return JSON with predefined items;
- fix JS docs in services.

File: app/service/file-service/index.js [hunk 3]
```diff
@@ -8,17 +9,34 @@ export class FileService {
    * Sends a request to get all the files from the server.
    *
    * @param {string} folderId - id of the folder, files from which should be requested.
-   * @return {Promise<unknown>} server response.
+   * @return {Promise<JSON>} server response.
```
Comment (dkalinin) [inline app/service/file-service/index.js]: Object.

File: app/store-configuration/action/index.js [hunk 4]
```diff
@@ -33,14 +62,14 @@ export default {
...
-          //  (server response).
-          store.commit(mutations.SET_CURRENT_FOLDER_ID, folderId);
-          return store.commit(mutations.FETCH_ITEMS, store.state.items);
+  [actions.FETCH_ITEMS](store) {
+    store.commit(mutations.SET_STATUS, 'Fetching items...');
+    return fileRequestService.getFiles(store.state.currentFolder.id)
+        .then((response) => {
+        // TODO:2019-09-04:bohdan.dudka: Second argument should be replaced with items
+        //  (server response).
+          store.commit(mutations.FETCH_ITEMS, response.items);
```
Comment (dkalinin) [inline app/store-configuration/action/index.js]: FETCH_ITEMS -> SET_ITEMS/FILES

File: app/view/component/file-list/index.js [hunk 5]
```diff
@@ -14,13 +14,12 @@ export class FileList extends DragAndDropComponent {
...
-  constructor(parent) {
+  constructor(parent, files) {
     super();
     this._root = parent;
-    this.subscribe('stateChange.status', () => {
-      this._loadSpan.textContent = this.state.status;
-    });
+    this._files = files;
     this.subscribe('stateChange.items', () => {
       this.render();
```
Comment (dkalinin) [inline app/view/component/file-list/index.js]: Please update `this._files` with new data from state.

File: app/view/component/file-list-item/index.js (hunk 1 above)
Comment (dkalinin) [reply app/view/component/file-list-item/index.js]: PTAL

**Comment by BohdanDudkaTD** [comment]
@dkalinin, PTAL, fixed issues:
- in JS docs replace `JSON` with `Object`;
- rename mutation `FETCH_ITEMS` to `SET_ITEMS`;
- in FileList update `this._files` each time a `stateChange.items` event occurs;
- add span into which name of the current folder is rendered. `FileListPage` is now subscribed for the `stateChange.currentFolder` event.

File: app/service/file-service/index.js [hunk 6]
```diff
@@ -82,11 +100,40 @@ export class FileService {
...
+  /**
+   * Sends a request to get model of the root folder.
+   *
+   * @return {Promise<Object>} model of the folder.
+   */
+  getRootFolder() {
+    return requestService.get('/folder/')
+        .then(() => {
+          return {
+            folder: new StorageFile('my-drive', 'my-drive', '2019-09-12', 'Bohdan', 'folder'),
```
Comment (dkalinin) [inline app/service/file-service/index.js]: Let's return StorageFile.

File: app/service/file-service/index.js [hunk 7]
```diff
@@ -82,11 +100,40 @@ export class FileService {
...
+   * Sends a request to get model of the folder by its id.
+   *
+   * @param {string} folderId - id of the folder which should be loaded.
+   * @return {Promise<Object>} model of the folder.
+   */
+  getFolderById(folderId) {
+    return requestService.get('/folder/' + folderId)
+        .then(() => {
+          return {
+            folder: new StorageFile(folderId, 'Folder_3', '2019-09-12', 'Bohdan', 'folder'),
```
Comment (dkalinin) [inline app/service/file-service/index.js]: Let's return StorageFile.

File: app/service/file-service/index.js [hunk 8]
```diff
@@ -8,17 +9,34 @@ export class FileService {
...
    * @param {string} folderId - id of the folder, files from which should be requested.
-   * @return {Promise<unknown>} server response.
+   * @return {Promise<Object>} server response.
    */
   getFiles(folderId) {
-    return requestService.get('/folder/' + folderId + '/files');
+    return requestService.get('/folder/' + folderId + '/files')
+        .then(() => {
+          return {
+            items: [
```
Comment (dkalinin) [inline app/service/file-service/index.js]: Let's return array.

**Comment by BohdanDudkaTD** [comment]
@dkalinin, @AndriiSameliuk, PTAL.
Fixed issues:
- lock `delete` button after it's clicked;
- in the `FileService` return already parsed objects;
- fix bug when details of the item are opened, another item is removed, an array with new objects is fetched but objects are now not equal when details are toggled.

File: app/view/page/file-list-page/index.js [hunk 9]
```diff
@@ -125,7 +128,8 @@ export class FileListPage extends ConfigurableComponent {
    * @private
    */
   _setSelectedFile(selectedFile) {
-    if (selectedFile === undefined || selectedFile === this._selectedFile) {
+    if (selectedFile === undefined
+      || JSON.stringify(selectedFile) === JSON.stringify(this._selectedFile)) {
```
Comment (dkalinin) [inline app/view/page/file-list-page/index.js]: Compare IDs of files.

**Pull request merged**